ecosystem/sep-0009.md

```diff
@@ -87,6 +87,14 @@ Name | Type | Description
 `organization.website` | string | Organization website
 `organization.email` | string | Organization contact email
 `organization.phone` | string | Organization contact phone
+`organization.individual_beneficial_owners.{index}.{SEP-0009 person field}` | array | Organization individual beneficial owners and control person
```

Comment:
This is the first case where a SEP-9 field contains an array with an index rather than sub-fields.

The SEP probably needs to specify some rules about how indexes and arrays work. Questions I have so far, which aren't exhaustive:
- Is zero or one the first index?
- Must indexes be sequential?
- Is an index forever? For example, if the list of beneficial owners changes from 4 persons to 3 persons, where of the 4, 2 are removed and 1 is added, how is that communicated in SEPs that use SEP-9? Do the removed persons indexes get reused? How do SEPs know that the person has changed vs a detail about a person has changed? Depending on the answers to these questions a unique identifier for a person may be preferred over an index.

Another thing we need to consider is that recently we had that case in another SEP where it became clear that encoding arrays in url-form-encoded format was ambiguous. We couldn't find any common way to encode the array where the index was preserved. URL encoding arrays using the only standard form does not traditionally preserve the order because there is no index encoded.

Other SEPs that use SEP-9 probably don't currently have instructions for how to use array elements, and we should include changes to those SEPs with this change otherwise it will be ambiguous for how to use and interpret then.

cc @JakeUrban

Reply:
SEPs we need to consider are: SEP-12, SEP-6, SEP-24, and soon SEP-8 once #758 merges.

Reply:
I think the best way to solve this problem is to avoid it entirely. That means using a unique identifier instead of array index as you suggest.

There are two alternatives:
- define our own standard for encoding arrays for key-value based content types: `application/x-www-form-urlencoded` and `multipart/form-data`
- use `application/json`. The blocker here is binary data, as JSON doesn't natively support it. However, we _could_ provide a standardized approach to encoding binary data in JSON: base64-encoding it.

To me, asking clients to base64-encode binary data seems easier than asking them to conform to a non-standard array-encoding scheme. I would also argue that there is precedent for using base64-encodings on binary data types within Stellar's ecosystem: `memo`.

Reply:
I like both approaches you discuss, and I think we could use both. Unique identifiers to make this problem mute, and supporting binary where-ever we use JSON via base64 encoding works. For the base64 encoding to work the field will also need to communicate the content-type of the contents. This could be in a sibling field with the same name suffixed with _content_type, or we could make the field in JSON format a JSON object with subfields `.content_type`, and `.data`.

Reply:
Don't we have the content type regardless of the base64 encoding? If an anchor requires `photo_id_front`, there's no way for the client to specify the image format used (png, jpeg, etc.). Why does encoding this data change that?

Reply:
We don't have the problem with multipart form encoding because each party can specify a `Content-Type` header, along with a `name` and even a `filename` field. This isn't shown in the examples of SEP-12, but it is available.

Reply:
Ah I see. I like the latter option more as well. I'll write up some PR's to add to each SEP's `Content-Type` section.

Reply:
used unique identifier to identify a beneficial owner. 
Do I need to specify what data type is allowed as unique identifier ? like is a Guid or a number or a string ? 
What normally you guys use for unique identifiers ?

Reply:
I think the unique identifier can be left undefined.